Alpaca.Markets/RestClient.cs

```diff
@@ -122,19 +152,46 @@ public void Dispose()
             for(var attempts = 0; attempts < throttler.MaxAttempts; ++attempts)
             {
                 throttler.WaitToProceed();
-
+                HttpStatusCode statusCode = HttpStatusCode.OK;
                 try
                 {
-                    using (var stream = await httpClient.GetStreamAsync(endpointUri))
-                    using (var reader = new JsonTextReader(new StreamReader(stream)))
+                    using (HttpResponseMessage response = await httpClient.GetAsync(endpointUri, HttpCompletionOption.ResponseHeadersRead))
                     {
-                        var serializer = new JsonSerializer();
-                        return serializer.Deserialize<TJson>(reader);
+                        // Adhere to server reported instructions
+                        statusCode = response.StatusCode;
+                        if (statusCode != HttpStatusCode.OK)
+                        {
+                            if (_retryHttpStatuses.Contains((int)statusCode))
+                            {
+                                // Accomodate Retry-After header
+                                var retryAfter = response.Headers.RetryAfter?.Delta?.TotalMilliseconds ?? (response.Headers.RetryAfter?.Date?.LocalDateTime - DateTime.Now)?.TotalMilliseconds;
+                                if (retryAfter != null)
+                                {
+                                    throttler.AllStop((Int32)retryAfter);
+                                }
+                                // Server unavailable, or Too many requests (429 can happen when this client competes with another client, e.g. mobile app)
+                                else if (statusCode == (HttpStatusCode)429 || statusCode == (HttpStatusCode)503)
```

Comment:
Logic error - if user do not add these codes into retryHttpStatuses list you will never hit this branch. Maybe you should add these codes into _retryHttpStatuses in any case?

Reply:
That's the intended behavior, I think he does not want to change the default behavior related to retries, just wants to provide a cleaner solution for handling them.

Reply:
OK. Will try to re-write problem. Do you agree that 'else if' branch will be executed only if `_retryHttpStatuses` contains value(s) 429 and/or 503? But it means what use should know about implementation details of this class. I suggest to add these 2 error code values into `_retryHttpStatuses` by default and provide proper retries handling even if user configures nothing.

Reply:
I see what you are saying, these are statuses that intrinsically indicate to the client to retry.  The caller shouldn't have to ask us to retry those, we should retry them because that is what a well-behaved Http client should do.

Reply:
Adjusted the logic to adhere to Retry-After headers, and any 429 or 503, as any well-behaved Http client should.  Beyond that the logic looks to the caller provided list of statuses to retry.